holoviews/core/data/multipath.py

```diff
@@ -0,0 +1,160 @@
+import numpy as np
+
+from ..util import max_range
+from .interface import Interface
+
+class MultiInterface(Interface):
+    """
+    MultiInterface allows wrapping around a list of tabular datasets
+    including dataframes, the columnar dictionary format or 2D tabular
+    NumPy arrays. Using the split method the list of tabular data can
+    be split into individual datasets.
+    """
+
+    types = ()
+
+    datatype = 'multi'
+
+    subtypes = ['dataframe', 'dictionary', 'array', 'dask']
+
+    @classmethod
+    def init(cls, eltype, data, kdims, vdims):
+        new_data = []
+        dims = {'kdims': eltype.kdims, 'vdims': eltype.vdims}
+        extra_kws = {}
+        for d in data:
+            d, interface, dims, extra_kws = Interface.initialize(eltype, d, kdims, vdims,
+                                                                 datatype=cls.subtypes)
+            new_data.append(d)
+        return new_data, dims, extra_kws
+
+    @classmethod
+    def validate(cls, dataset):
+        pass
+
+    @classmethod
+    def template(cls, dataset):
+        """
+        Returns a Dataset template used as a wrapper around the data
+        contained within the multi-interface dataset.
+        """
+        from . import Dataset
+        vdims = dataset.vdims if getattr(dataset, 'level', None) is None else []
+        return Dataset(dataset.data[0], datatype=cls.subtypes,
+                       kdims=dataset.kdims, vdims=vdims)
+
+    @classmethod
+    def dimension_type(cls, dataset, dim):
+        if not dataset.data:
+            return float
+        ds = cls.template(dataset)
+        return ds.interface.dimension_type(ds, dim)
+
+    @classmethod
+    def range(cls, dataset, dim):
+        if not dataset.data:
+            return (None, None)
+        ranges = []
+        ds = cls.template(dataset)
+
+        # Backward compatibility for level
+        level = getattr(dataset, 'level', None)
+        dim = dataset.get_dimension(dim)
+        if level is not None and dim is dataset.vdims[0]:
+            return (level, level)
+
+        for d in dataset.data:
+            ds.data = d
+            ranges.append(ds.interface.range(ds, dim))
+        return max_range(ranges)
+
+    @classmethod
+    def select(cls, dataset, selection_mask=None, **selection):
```

Comment:
Right, although it's not exposed at the ``Path`` level because we didn't allow slicing paths in the past. Will have to decide what to do here in future, in geopandas interface doing it properly will be trivial but handling polygon slicing correctly here would be a pain.